sdk/python/feast/client.py

```diff
@@ -353,6 +360,145 @@ def archive_project(self, project):
         if self._project == project:
             self._project = FEAST_DEFAULT_OPTIONS[CONFIG_PROJECT_KEY]
 
+    def apply_entity(
+        self, entities: Union[List[EntityV2], EntityV2], project: str = None
+    ):
+        """
+        Idempotently registers entities with Feast Core. Either a single
+        entity or a list can be provided.
+
+        Args:
+            entities: List of entities that will be registered
+
+        Examples:
+            >>> from feast import Client
+            >>> from feast.entity import EntityV2
+            >>> from feast.value_type import ValueType
+            >>>
+            >>> feast_client = Client(core_url="localhost:6565")
+            >>> entity = EntityV2(
+            >>>     name="driver_entity",
+            >>>     description="Driver entity for car rides",
+            >>>     value_type=ValueType.STRING,
+            >>>     labels={
+            >>>         "key": "val"
+            >>>     }
+            >>> )
+            >>> feast_client.apply_entity(entity)
+        """
+
+        if project is None:
+            if self.project is not None:
```

Comment:
Can we move this if/else into the project property so that the ValueError is thrown there? Otherwise people will continue to have to implement this conditional. `self.property` should always return something or throw an exception.

Reply:
Updated.